### PR TITLE
Fix generic json schema for scala3

### PR DESCRIPTION
**Pull request by mleclercq**
Addresses https://github.com/endpoints4s/endpoints4s/issues/566#issuecomment-1052136445

### CODE REVIEW

File: json-schema/json-schema-generic/src/test/scala/endpoints4s/generic/JsonSchemasCirceTest.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package endpoints4s
+package generic
+
+import io.circe.Json
+import org.scalatest.freespec.AnyFreeSpec
+
+class JsonSchemasCirceTest extends AnyFreeSpec {
+
+  object GenecirCodec extends JsonSchemasDocs with JsonSchemas with circe.JsonSchemas
```
Comment (julienrf) [inline json-schema/json-schema-generic/src/test/scala/endpoints4s/generic/JsonSchemasCirceTest.scala:9]: There is a typo here

File: json-schema/json-schema-generic/src/test/scala/endpoints4s/generic/JsonSchemasCirceTest.scala [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package endpoints4s
+package generic
+
+import io.circe.Json
+import org.scalatest.freespec.AnyFreeSpec
+
+class JsonSchemasCirceTest extends AnyFreeSpec {
```
Comment (julienrf) [inline json-schema/json-schema-generic/src/test/scala/endpoints4s/generic/JsonSchemasCirceTest.scala:7]: Do the other (existing) tests cross-compile? What is the purpose of this new test?

**Review by julienrf — COMMENTED** [review]
Thank you so much @mleclercq!

That looks great!

File: json-schema/json-schema-generic/src/test/scala/endpoints4s/generic/JsonSchemasCirceTest.scala (hunk 2 above)
Comment (mleclercq) [reply json-schema/json-schema-generic/src/test/scala/endpoints4s/generic/JsonSchemasCirceTest.scala:7]: > Do the other (existing) tests cross-compile?

I'm not sure what you mean. AFAIK all tests in this module compiles and pass on both scala 2 and 3

> What is the purpose of this new test?

I wanted to make sure the actual encoding/decoding was working especially given all the `asInstanceOf` being done and to make sure that I didn't get an "off by one" kind of error in the change I did regarding the ordinal in `summonTagged`.

While `JsonSchemasTest` tests the generic derivation of the codecs. AFAIU, it doesn't really encode/decode anything so all the code inside the `xmap` was not tested (as shown by the coverage report)

File: json-schema/json-schema-generic/src/test/scala/endpoints4s/generic/JsonSchemasCirceTest.scala (hunk 1 above)
Comment (mleclercq) [reply json-schema/json-schema-generic/src/test/scala/endpoints4s/generic/JsonSchemasCirceTest.scala]: Fixed

File: json-schema/json-schema-generic/src/test/scala/endpoints4s/generic/JsonSchemasCirceTest.scala (hunk 2 above)
Comment (julienrf) [reply json-schema/json-schema-generic/src/test/scala/endpoints4s/generic/JsonSchemasCirceTest.scala:7]: Yeah, I didn’t remember which tests were cross-compiled and cross-ran.

OK, thank you for expanding the test coverage. I don’t have much time to improve the test suite for now, feel free to add more tests if needed!

**Review by julienrf — APPROVED** [review]

**Pull request merged**